From: Anton Arapov <aarapov@redhat.com>
Date: Thu, 11 Dec 2008 15:09:28 +0100
Subject: [misc] futex.h: remove kernel bits for userspace header
Message-id: 20081211140927.GE3411@redhat.com
O-Subject: [RHEL5.4 PATCH] BZ475790: Fix noise in futex.h
Bugzilla: 475790
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: John Feeney <jfeeney@redhat.com>

Bugzilla: 475790

Description:
  There are some kernel-only bits in the middle of <linux/futex.h>
which should be removed in what we export to userspace.

Upstream: #58f64d83c37f5073a01573d27043c9c0ccc764f1

Please, ACK. :)

==

 include/linux/Kbuild  |    2 +-
 include/linux/futex.h |    2 ++
 2 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/include/linux/Kbuild b/include/linux/Kbuild
index e7403e6..b507e6e 100644
--- a/include/linux/Kbuild
+++ b/include/linux/Kbuild
@@ -67,7 +67,6 @@ header-y += ftape-vendors.h
 header-y += firewire-cdev.h
 header-y += firewire-constants.h
 header-y += fuse.h
-header-y += futex.h
 header-y += genetlink.h
 header-y += gen_stats.h
 header-y += gigaset_dev.h
@@ -209,6 +208,7 @@ unifdef-y += fb.h
 unifdef-y += fcntl.h
 unifdef-y += filter.h
 unifdef-y += flat.h
+unifdef-y += futex.h
 unifdef-y += fs.h
 unifdef-y += ftape.h
 unifdef-y += gameport.h
diff --git a/include/linux/futex.h b/include/linux/futex.h
index 7e9abdc..23cec9e 100644
--- a/include/linux/futex.h
+++ b/include/linux/futex.h
@@ -105,6 +105,7 @@ struct robust_list_head {
  */
 #define ROBUST_LIST_LIMIT	2048
 
+#ifdef __KERNEL__
 long do_futex(u32 __user *uaddr, int op, u32 val, unsigned long timeout,
 	      u32 __user *uaddr2, u32 val2, u32 val3);
 
@@ -122,6 +123,7 @@ static inline void exit_pi_state_list(struct task_struct *curr)
 {
 }
 #endif
+#endif /* __KERNEL__ */
 
 #define FUTEX_OP_SET		0	/* *(int *)UADDR2 = OPARG; */
 #define FUTEX_OP_ADD		1	/* *(int *)UADDR2 += OPARG; */
